integrationtests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -384,6 +384,23 @@ void requestFiltersCanRespondWithoutProxyingRespondsInCorrectOrder() throws Exce
         }
     }
 
+    @Test
+    void clientsCanSendMultipleMessagesImmediately() throws Exception {
+
+        try (var tester = mockKafkaKroxyliciousTester(s -> proxy(s).addToFilters(new FilterDefinitionBuilder("RejectingCreateTopic").build()));
+                var client = tester.simpleTestClient()) {
+            tester.addMockResponseForApiKey(new ResponsePayload(METADATA, METADATA.latestVersion(), new MetadataResponseData()));
+            Request requestA = new Request(METADATA, METADATA.latestVersion(), "clientA", new MetadataRequestData());
+            Request requestB = new Request(METADATA, METADATA.latestVersion(), "clientB", new MetadataRequestData());
+            var futureA = client.get(requestA);
+            var futureB = client.get(requestB);
+            Response responseA = futureA.get(10, TimeUnit.SECONDS);
+            Response responseB = futureB.get(10, TimeUnit.SECONDS);
+            assertThat(responseA.payload()).isNotNull();
```

Comment:
should add some way of disambiguating responseA and responseB, maybe use different request types